src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs

```diff
@@ -43,7 +49,15 @@ public static IResourceBuilder<NatsServerResource> AddNats(this IDistributedAppl
                 LoggerFactory = @event.Services.GetRequiredService<ILoggerFactory>(),
             };
 
-            options = options with { Url = connectionString };
+            options = options with
+            {
+                Url = connectionString,
+                AuthOpts = new()
+                {
+                    Username = await nats.UserNameReference.GetValueAsync(ct).ConfigureAwait(false),
+                    Password = nats.PasswordParameter!.Value,
```

Comment:
this is good correct way of passing user, pass to the client. no need to have it in the url